src/Neo/SmartContract/Native/RoleManagement.cs

```diff
@@ -73,9 +73,8 @@ private void DesignateAsRole(ApplicationEngine engine, Role role, ECPoint[] node
         list.AddRange(nodes);
         list.Sort();
         engine.SnapshotCache.Add(key, new StorageItem(list));
-        var oldNodes = new VM.Types.Array(engine.ReferenceCounter, GetDesignatedByRole(engine.SnapshotCache, role, index - 1).Select(u => (ByteString)u.EncodePoint(true)));
-        var newNodes = new VM.Types.Array(engine.ReferenceCounter, nodes.Select(u => (ByteString)u.EncodePoint(true)));
-        engine.SendNotification(Hash, "Designation", new VM.Types.Array(engine.ReferenceCounter, [(int)role, engine.PersistingBlock.Index, oldNodes, newNodes]));
+        var oldNodes = GetDesignatedByRole(engine.SnapshotCache, role, index - 1);
+        Notify(engine, "Designation", role, engine.PersistingBlock.Index, oldNodes, nodes);
```

Comment:
https://github.com/neo-project/neo/blob/46bf558f64a8e1045cab8789890aa1a0d242ec5b/src/Neo/SmartContract/ApplicationEngine.cs#L506

Seems that convert don't check if it's a publicKey, so it will produce undesired results